packages/react-form-renderer/src/form-renderer/condition.js

```diff
@@ -27,48 +29,137 @@ const fieldCondition = (value, { is, isNotEmpty, isEmpty, pattern, notMatch, fla
+    return !condition.and.map((condition) => parseCondition(condition, values)).some(({ result }) => result === false)
+      ? positiveResult
+      : negativeResult;
+  }
+
+  if (condition.sequence) {
+    return condition.sequence.reduce(
+      (acc, curr) => {
+        const result = parseCondition(curr, values);
+
+        return {
+          sets: [...acc.sets, ...(result.set ? [result.set] : [])],
+          visible: acc.visible || result.visible,
+          result: acc.result || result.result
+        };
+      },
+      { ...negativeResult, sets: [] }
+    );
   }
 
   if (condition.or) {
-    return condition.or.map((condition) => parseCondition(condition, values)).some((result) => result === true);
+    return condition.or.map((condition) => parseCondition(condition, values)).some(({ result }) => result === true) ? positiveResult : negativeResult;
   }
 
   if (condition.not) {
-    return !parseCondition(condition.not, values);
+    return !parseCondition(condition.not, values).result ? positiveResult : negativeResult;
   }
 
   if (typeof condition.when === 'string') {
-    return fieldCondition(get(values, condition.when), condition);
+    return fieldCondition(get(values, condition.when), condition) ? positiveResult : negativeResult;
   }
 
   if (Array.isArray(condition.when)) {
-    return !!condition.when.map((fieldName) => fieldCondition(get(values, fieldName), condition)).find((condition) => !!condition);
+    return condition.when.map((fieldName) => fieldCondition(get(values, fieldName), condition)).find((condition) => !!condition)
+      ? positiveResult
+      : negativeResult;
   }
 
-  return false;
+  return negativeResult;
 };
 
-const Condition = ({ condition, children }) => <FormSpy>{({ values }) => (parseCondition(condition, values) ? children : null)}</FormSpy>;
+const Condition = React.memo(
+  ({ condition, children, values }) => {
+    const formOptions = useFormApi();
+    const dirty = formOptions.getState().dirty;
+
+    const [lastSets, setSets] = React.useState([]);
+    const [initial, setInitial] = React.useState(true);
+
+    const conditionResult = parseCondition(condition, values, formOptions);
+    const setters = conditionResult.set ? [conditionResult.set] : conditionResult.sets;
+
+    useEffect(() => {
+      if (!dirty) {
+        setInitial(true);
+      }
+    }, [dirty]);
+
+    useEffect(() => {
+      if (setters && setters.length > 0 && (initial || !isEqual(setters, lastSets))) {
+        setters.forEach((setter, index) => {
+          if (setter && (initial || !isEqual(setter, lastSets[index]))) {
+            setTimeout(() => {
+              formOptions.batch(() => {
+                Object.entries(setter).forEach(([name, value]) => {
+                  formOptions.change(name, value);
+                });
+              });
+            });
+          }
+        });
+        setSets(setters);
```

Comment:
we are doing two setters in one effect, that will cause additional render. Can we move from `useState` to `useReducer`?